<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainView</class>
 <widget class="QMainWindow" name="MainView">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>802</width>
    <height>755</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>802</width>
    <height>755</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>802</width>
    <height>755</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>MLP</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="s21::Drawer" name="drawer">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>512</width>
      <height>512</height>
     </rect>
    </property>
    <property name="mouseTracking">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QLabel" name="result_label">
    <property name="geometry">
     <rect>
      <x>530</x>
      <y>10</y>
      <width>80</width>
      <height>80</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>72</pointsize>
     </font>
    </property>
    <property name="frameShape">
     <enum>QFrame::NoFrame</enum>
    </property>
    <property name="text">
     <string>A</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QPushButton" name="clear_button">
    <property name="geometry">
     <rect>
      <x>90</x>
      <y>530</y>
      <width>120</width>
      <height>50</height>
     </rect>
    </property>
    <property name="text">
     <string>Очистить</string>
    </property>
   </widget>
   <widget class="QPushButton" name="train_button">
    <property name="geometry">
     <rect>
      <x>570</x>
      <y>100</y>
      <width>161</width>
      <height>51</height>
     </rect>
    </property>
    <property name="text">
     <string>Обучить</string>
    </property>
   </widget>
   <widget class="QPushButton" name="open_image_button">
    <property name="geometry">
     <rect>
      <x>280</x>
      <y>530</y>
      <width>161</width>
      <height>50</height>
     </rect>
    </property>
    <property name="text">
     <string>Открыть изображение</string>
    </property>
   </widget>
   <widget class="QPushButton" name="test_button">
    <property name="geometry">
     <rect>
      <x>570</x>
      <y>220</y>
      <width>161</width>
      <height>50</height>
     </rect>
    </property>
    <property name="text">
     <string>Протестировать</string>
    </property>
   </widget>
   <widget class="QPushButton" name="save_weights_button">
    <property name="geometry">
     <rect>
      <x>570</x>
      <y>280</y>
      <width>161</width>
      <height>50</height>
     </rect>
    </property>
    <property name="text">
     <string>Сохранить веса</string>
    </property>
   </widget>
   <widget class="QPushButton" name="load_weights_button">
    <property name="geometry">
     <rect>
      <x>570</x>
      <y>340</y>
      <width>161</width>
      <height>50</height>
     </rect>
    </property>
    <property name="text">
     <string>Загрузить веса</string>
    </property>
   </widget>
   <widget class="QFrame" name="test_results_frame">
    <property name="geometry">
     <rect>
      <x>530</x>
      <y>410</y>
      <width>261</width>
      <height>301</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::StyledPanel</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Raised</enum>
    </property>
    <widget class="QLabel" name="frame_title">
     <property name="geometry">
      <rect>
       <x>40</x>
       <y>10</y>
       <width>141</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>Результаты теста</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QWidget" name="verticalLayoutWidget">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>40</y>
       <width>241</width>
       <height>251</height>
      </rect>
     </property>
     <layout class="QVBoxLayout" name="test_results_layout">
      <property name="sizeConstraint">
       <enum>QLayout::SetDefaultConstraint</enum>
      </property>
      <property name="leftMargin">
       <number>10</number>
      </property>
      <property name="rightMargin">
       <number>10</number>
      </property>
      <item>
       <layout class="QHBoxLayout" name="avg_acc_layout">
        <item>
         <widget class="QLabel" name="avg_acc_label">
          <property name="text">
           <string>Средняя точность</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QDoubleSpinBox" name="avg_acc_spinbox">
          <property name="buttonSymbols">
           <enum>QAbstractSpinBox::NoButtons</enum>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="precision_layout">
        <item>
         <widget class="QLabel" name="precision_label">
          <property name="text">
           <string>Прецизионность</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QDoubleSpinBox" name="precision_spinbox">
          <property name="buttonSymbols">
           <enum>QAbstractSpinBox::NoButtons</enum>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="recall_layout">
        <item>
         <widget class="QLabel" name="recall_label">
          <property name="text">
           <string>Полнота</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QDoubleSpinBox" name="recall_spinbox">
          <property name="buttonSymbols">
           <enum>QAbstractSpinBox::NoButtons</enum>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="f_measure_layout">
        <item>
         <widget class="QLabel" name="f_measure_label">
          <property name="text">
           <string>F - мера</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QDoubleSpinBox" name="f_measure_spinbox">
          <property name="buttonSymbols">
           <enum>QAbstractSpinBox::NoButtons</enum>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="elapsed_time_layout">
        <item>
         <widget class="QLabel" name="elapsed_time_label">
          <property name="text">
           <string>Затраченное время</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QDoubleSpinBox" name="elapsed_time_spinbox">
          <property name="buttonSymbols">
           <enum>QAbstractSpinBox::NoButtons</enum>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label">
          <property name="text">
           <string>с.</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </widget>
   <widget class="QGroupBox" name="perceptron_type_group">
    <property name="geometry">
     <rect>
      <x>130</x>
      <y>600</y>
      <width>261</width>
      <height>81</height>
     </rect>
    </property>
    <property name="title">
     <string>Реализация перцептрона</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
    <property name="flat">
     <bool>true</bool>
    </property>
    <property name="checkable">
     <bool>false</bool>
    </property>
    <property name="checked">
     <bool>false</bool>
    </property>
    <widget class="QWidget" name="horizontalLayoutWidget">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>20</y>
       <width>261</width>
       <height>61</height>
      </rect>
     </property>
     <layout class="QHBoxLayout" name="realization_layout">
      <item>
       <widget class="QRadioButton" name="matrix_type_button">
        <property name="text">
         <string>Матричная</string>
        </property>
        <property name="checkable">
         <bool>true</bool>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QRadioButton" name="graph_type_button">
        <property name="text">
         <string>Графовая</string>
        </property>
        <property name="checkable">
         <bool>true</bool>
        </property>
        <property name="checked">
         <bool>false</bool>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </widget>
   <widget class="QPushButton" name="cross_validation_button">
    <property name="geometry">
     <rect>
      <x>570</x>
      <y>160</y>
      <width>161</width>
      <height>51</height>
     </rect>
    </property>
    <property name="text">
     <string>Кросс-валидация</string>
    </property>
   </widget>
  </widget>
 </widget>
 <customwidgets>
  <customwidget>
   <class>s21::Drawer</class>
   <extends>QGraphicsView</extends>
   <header>View/drawer.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
